src/SearchField/SearchFieldSubmitButton.jsx

```diff
@@ -1,21 +1,72 @@
 import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
 
 import { SearchFieldContext } from './SearchFieldAdvanced';
+import { Button } from '../index';
+
+const STYLE_VARIANTS = [
+  'light',
+  'dark',
+];
+
+const BUTTON_LOCATION_VARIANTS = [
+  'internal',
+  'external',
+];
 
 const SearchFieldSubmitButton = (props) => {
-  const { screenReaderText, icons, refs } = useContext(SearchFieldContext);
+  const {
+    variant, submitButtonLocation, buttonText, ...others
+  } = props;
+  const {
+    screenReaderText, icons, refs, value,
+  } = useContext(SearchFieldContext);
+
+  if (submitButtonLocation === 'internal' && value.length) {
+    return null;
+  }
 
-  return (
+  return submitButtonLocation === 'external' ? (
+    <Button
+      type="submit"
+      ref={refs.submitButton}
+      variant={variant === 'light' ? 'dark' : 'danger'}
```

Comment:
Should these be `primary` and `brand` (per Figma)?

Reply:
Yes, I will update it